From 676ae7528dbae4b3997293f7ec876bda435a76db Mon Sep 17 00:00:00 2001
From: Jianwei Hu <jiahu@redhat.com>
Date: Fri, 5 Dec 2014 14:26:22 +0800
Subject: [PATCH 13/13] Remove extraneous single quotes from IPV6ADDR_SECONDARIES

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1208894

The IPV6ADDR_SECONDARIES setting in ifcfg files was being set
with an extra set of single quotes.

For example:
After defining a interface with multi-ipv6 addresses
[root@localhost ~]# cat /etc/sysconfig/network-scripts/ifcfg-enp2s0
DEVICE=enp2s0
ONBOOT=yes
IPV6INIT=yes
IPV6_AUTOCONF=no
DHCPV6C=no
IPV6ADDR=2002::200/64
IPV6ADDR_SECONDARIES="'2002::201/64 2002::202/64'"

It should instead look like this:
[root@localhost ~]# cat /etc/sysconfig/network-scripts/ifcfg-enp2s0
DEVICE=enp2s0
ONBOOT=yes
IPV6INIT=yes
IPV6_AUTOCONF=no
DHCPV6C=no
IPV6ADDR=2002::200/64
IPV6ADDR_SECONDARIES="2002::201/64 2002::202/64"

Since the extra quotes are added in all cases, this patch just
unconditionally removes them.
(cherry picked from commit 597032732156e03af15bdbd0c4a0e7a89e66a8f3)
---
 data/xml/redhat-get.xsl                   |    2 --
 data/xml/suse-get.xsl                     |    2 --
 tests/redhat/schema/ipv6-static-multi.xml |    2 +-
 tests/suse/schema/ipv6-static-multi.xml   |    2 +-
 4 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/data/xml/redhat-get.xsl b/data/xml/redhat-get.xsl
index 370f4ca..cc73f69 100644
--- a/data/xml/redhat-get.xsl
+++ b/data/xml/redhat-get.xsl
@@ -227,14 +227,12 @@
     <xsl:if test="count(ip) > 1">
       <node label="IPV6ADDR_SECONDARIES">
         <xsl:attribute name="value">
-          <xsl:text>'</xsl:text>
           <xsl:for-each select="ip[1]/following-sibling::ip[following-sibling::ip]">
             <xsl:value-of select="@address"/><xsl:if test="@prefix">/<xsl:value-of select="@prefix"/></xsl:if><xsl:value-of select="string(' ')"/>
           </xsl:for-each>
           <xsl:for-each select="ip[last()]">
             <xsl:value-of select="@address"/><xsl:if test="@prefix">/<xsl:value-of select="@prefix"/></xsl:if>
           </xsl:for-each>
-          <xsl:text>'</xsl:text>
         </xsl:attribute>
       </node>
     </xsl:if>
diff --git a/data/xml/suse-get.xsl b/data/xml/suse-get.xsl
index 7336616..0873f42 100644
--- a/data/xml/suse-get.xsl
+++ b/data/xml/suse-get.xsl
@@ -244,14 +244,12 @@
     <xsl:if test="count(ip) > 1">
       <node label="IPV6ADDR_SECONDARIES">
         <xsl:attribute name="value">
-          <xsl:text>'</xsl:text>
           <xsl:for-each select="ip[1]/following-sibling::ip[following-sibling::ip]">
             <xsl:value-of select="@address"/><xsl:if test="@prefix">/<xsl:value-of select="@prefix"/></xsl:if><xsl:value-of select="string(' ')"/>
           </xsl:for-each>
           <xsl:for-each select="ip[last()]">
             <xsl:value-of select="@address"/><xsl:if test="@prefix">/<xsl:value-of select="@prefix"/></xsl:if>
           </xsl:for-each>
-          <xsl:text>'</xsl:text>
         </xsl:attribute>
       </node>
     </xsl:if>
diff --git a/tests/redhat/schema/ipv6-static-multi.xml b/tests/redhat/schema/ipv6-static-multi.xml
index d865a05..f1f82cd 100644
--- a/tests/redhat/schema/ipv6-static-multi.xml
+++ b/tests/redhat/schema/ipv6-static-multi.xml
@@ -7,6 +7,6 @@
     <node label="IPV6_AUTOCONF" value="no"/>
     <node label="DHCPV6C" value="no"/>
     <node label="IPV6ADDR" value="3ffe:ffff:0:5::1/128"/>
-    <node label="IPV6ADDR_SECONDARIES" value="'3ffe:ffff:0:5::3/128 3ffe:ffff:0:5::5/128'"/>
+    <node label="IPV6ADDR_SECONDARIES" value="3ffe:ffff:0:5::3/128 3ffe:ffff:0:5::5/128"/>
   </tree>
 </forest>
diff --git a/tests/suse/schema/ipv6-static-multi.xml b/tests/suse/schema/ipv6-static-multi.xml
index d865a05..f1f82cd 100644
--- a/tests/suse/schema/ipv6-static-multi.xml
+++ b/tests/suse/schema/ipv6-static-multi.xml
@@ -7,6 +7,6 @@
     <node label="IPV6_AUTOCONF" value="no"/>
     <node label="DHCPV6C" value="no"/>
     <node label="IPV6ADDR" value="3ffe:ffff:0:5::1/128"/>
-    <node label="IPV6ADDR_SECONDARIES" value="'3ffe:ffff:0:5::3/128 3ffe:ffff:0:5::5/128'"/>
+    <node label="IPV6ADDR_SECONDARIES" value="3ffe:ffff:0:5::3/128 3ffe:ffff:0:5::5/128"/>
   </tree>
 </forest>
-- 
1.7.1

